WordPress/gutenberg
# Block Bindings: Bootstrap sources defined in the server

## What?
This PR bootstraps the block bindings sources defined in the server so some of the properties are available without needing to register them in the client.

It follows a similar approach to `addBootstrappedBlockType`.

## Why?
It reduces complexity for developers and let us reuse the server registration.

## How?
* Add a compat filter to expose the sources in the editor settings from the server. (I created a related [pull request in core](https://github.com/WordPress/wordpress-develop/pull/7020) as well).
* Slightly change the registration to understand that a source could be bootstraped from the server.
* Create a new `addBootstrappedBlockBindingsSource` action to initialize the server sources with the minimal information.
* Add a new `bootstrapBlockBindingsSourcesFromServer` that receives the editor settings and bootstraps the sources.
* Call this function from `edit-site` and `edit-post` at the same time blocks and other block bindings sources are registered.

## Testing Instructions
1. Register a new source from the server. You can use something like this:
```
function init_source() {
	register_block_bindings_source(
		'plugin/custom',
		array(
			'label'              => _x( 'Custom Source', 'block bindings source' ),
			'get_value_callback' => function() {},
			'uses_context'       => array( 'postId', 'postType' ),
		)
	);
}

add_action( 'init', 'init_source' );
```

2. Go to a page and a paragraph connected to your source. Something like this:

```

<p>Default content</p>

```

3. Check that the label is shown instead of the name in the connections panel in the right sidebar.
